src/Redirect.tsx

```diff
@@ -1,13 +1,19 @@
+/* eslint-disable max-classes-per-file */
+import React from 'react';
+
 import RedirectException from './RedirectException';
-import { Match, RedirectOptions } from './typeUtils';
+import { LocationDescriptor, Match, RedirectOptions } from './typeUtils';
+
+class Redirect implements RedirectOptions {
+  path?: string;
+
+  to: string | ((match: Match) => LocationDescriptor);
+
+  status?: number;
```

Comment:
I decided to just declare those on the classes. Technically those are functional changes, but pracically they should not change anything. I'm pretty sure the only difference is that `this` has those initialized as `undefined` when creating the instance, which should not change anything